querybuilder/query.py

```diff
@@ -1426,6 +1426,17 @@ def wrap(self, alias=None):
 
         return self
 
+    def copy(self):
+        """
+        Deeply copies everything in the query object and safely copies the connection object
+        """
+        connection = self.connection
+        del self.connection
+        copied_query = deepcopy(self)
+        copied_query.connection = connection
+        self.connection = connection
+        return copied_query
```

Comment:
deepcopy was causing an error when trying to copy the connection object. not sure if there is a better way to handle this.